site-src/v1alpha2/guides/http-redirect-rewrite.md

```diff
@@ -0,0 +1,180 @@
+          scheme: https
+		  statusCode: 301
+      backendRefs:
+      - name: example-svc
+        weight: 1
+        port: 80
+```
+
+Redirects change configured URL components to match the redirect configuration
+while preserving other components from the original request URL. In this
+example, the request `GET http://redirect.example/cinammon` will result in a
+301 response with a `location: https://redirect.example/cinnamon` header. The
+hostname (`redirect.example`), path (`/cinnamon`), and port (implicit) remain
+unchanged.
+
+<!---
+The above raises some questions. Do redirects apply unconditionally? If my
+request is already HTTPS, will the above create a redirect loop? If redirects
+are unconditional (i.e. they _do not_ nothing and proxy upstream if the request
+already uses HTTPS), how should you properly write configuration elsewhere to
+ensure this only applies to HTTP requests?
+-->
+
+### Path redirects
+
+Path redirects use an HTTP Path Modifier to replace either entire paths or path
+prefixes. For example, 
+
+```
+apiVersion: gateway.networking.k8s.io/v1alpha2
+kind: HTTPRoute
+metadata:
+  name: http-filter-redirect
+spec:
+  hostnames:
+    - redirect.example
+  rules:
+    - matches:
+      - path:
+          type: PathPrefix
+          value: /cayenne
+      filters:
+      - type: RequestRedirect
+	    requestRedirect:
+          path:
+		    type: ReplaceFullPath
+			replaceFullPath: /paprika
+		  statusCode: 302
+      backendRefs:
+      - name: example-svc
+        weight: 1
+        port: 80
+```
+
+will issue a 302 redirect to all `redirect.example` requests whose path begins
+with `/cayenne` to `/paprika`. Both requests to
+`https://redirect.example/cayenne/pinch` and
+`https://redirect.example/cayenne/teaspoon` will receive a redirect with a
+`location: https://redirect.example/paprika`.
+
+The other path redirect type, `ReplacePrefixMatch`, replaces only the path
+portion matching `matches.path.value`. Changing the filter in the above to:
+
+```
+      - type: RequestRedirect
+	    requestRedirect:
+          path:
+		    type: ReplacePrefixMatch
+			replacePrefixMatch: /paprika
+		  statusCode: 302
+```
+
+will result in redirects with `location:
+https://redirect.example/paprika/pinch` and `location:
+https://redirect.example/paprika/teaspoon` response headers.
+
+<!---
+Do these behave at all differently based on matches.path.type? I expect no, and
+that implementations should just always replace only the matching path prefix
+or the entire path. Using ReplacePrefixMatch on Exact is the same as
+ReplaceFullPath. RegularExpression looks a bit ambiguous, since the
+PathMatchType looks unclear on whether they're handled as prefixes or not.
+-->
+
+## Rewrites
+
+Rewrites modify components of a client request before proxying it upstream. A
+[`URLRewrite`
+filter](/v1alpha2/references/spec/#gateway.networking.k8s.io/v1alpha2.HTTPURLRewriteFilter)
+can change the upstream request hostname and/or path. For example,
```

Comment:
I think this might read better if you merged the content from before and after the example, something like this:
```suggestion
can change the upstream request hostname and/or path. For example, the following HTTPRoute will accept...
```

Same comment applies elsewhere.